pkg/cmd/playback.go

```diff
@@ -0,0 +1,245 @@
+		// This flag doesn't actually hide the command because we use a template to display the --help text
+		// To unhide this command, see getUsageTemplate() in templates.go, where we explicitly hide the `playback` cmd.
+		Hidden: true,
+		Use:    "playback",
+		Args:   validators.NoArgs,
+		Short:  "Start a `playback` server",
+		Long: `
+--- Overview ---
+The playback command starts a local proxy server that intercepts outgoing requests to the Stripe API.
+
+There are three modes of operation:
+
+"record": Any requests received are forwarded to the api.stripe.com, and the response is returned. All interactions
+are written to the loaded 'cassette' file for later playback in replay mode.
+
+"replay": All received requests are terminated at the playback server, and responses are played back[1] from a cassette file. A existing cassette most be loaded.
+
+"auto": The server determines whether to run in "record" or "replay" mode on a per-cassette basis. If the cassette exists, operates in "replay" mode. If not, operates in "record" mode.
+
+Currently, stripe playback only supports serving over HTTP.
+
+[1]: requests are currently replayed sequentially in the same order they were recorded.
+` + endpointsDocString,
+		Example: `stripe playback
+  stripe playback --replaymode
+  stripe playback --cassette "my_cassette.yaml"`,
+		RunE: pc.runPlaybackCmd,
+	}
+
+	pc.cmd.Flags().StringVar(&pc.mode, "mode", "auto", "Auto: record if cassette doesn't exist, replay if exists. Record: always record/re-record. Replay: always replay.")
+	pc.cmd.Flags().StringVar(&pc.address, "address", fmt.Sprintf("localhost:%d", defaultPort), "Address to serve on")
+	pc.cmd.Flags().StringVar(&pc.webhookAddress, "forward-to", fmt.Sprintf("localhost:%d", defaultWebhookPort), "Address to forward webhooks to")
+	pc.cmd.Flags().StringVar(&pc.filepath, "cassette", "default_cassette.yaml", "The cassette file to use")
+	pc.cmd.Flags().StringVar(&pc.cassetteDir, "cassette-root-dir", "", "Directory to store all cassettes in. All cassette paths are interpreted as relative to this directory.")
+	// pc.cmd.Flags().BoolVar(&pc.serveHTTPS, "https", false, "Serve over HTTPS (default: HTTP")
+
+	// // Hidden configuration flags, useful for dev/debugging
+	pc.cmd.Flags().StringVar(&pc.apiBaseURL, "api-base", "https://api.stripe.com", "Sets the API base URL")
+
+	return pc
+}
+
+func (pc *playbackCmd) runPlaybackCmd(cmd *cobra.Command, args []string) error {
+	fmt.Println()
+	fmt.Println("Seting up playback server...")
+	fmt.Println()
+
+	// --- Validate command-line args
+	// Check that mode is valid
+	if pc.mode != playback.Auto && pc.mode != playback.Record && pc.mode != playback.Replay {
+		return fmt.Errorf(
+			"\"%v\" is not a valid mode. It must be either \"%v\", \"%v\", or \"%v\"",
+			pc.mode, playback.Auto, playback.Record, playback.Replay)
+	}
+
+	// Check that cassette root directory is valid
+	var absoluteCassetteDir string
+	var err error
+	if filepath.IsAbs(pc.cassetteDir) {
```

Comment:
Do we need this? `filePath.Abs` docs sounds like it noops if the path is already absolute? https://golang.org/pkg/path/filepath/#Abs

Reply:
You're right, will fix!